docs/DeveloperGuide.md

```diff
@@ -8,6 +8,153 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Updating a Task
+
+### Overview
+
+Updating a task involves modifying its description or other attributes such as date or time. This guide outlines the process and considerations for updating tasks within the application.
+
+### Update Manager Method
+
+The `updateManager` method facilitates the updating of tasks. It prompts the user for an updated task description and handles tasks of different types (Todo, Event, Deadline) while ensuring the changes are correctly reflected in the application.
+
+#### Method Signature
+
+```java
+public void updateManager(Scanner scanner, WeekView weekView, MonthView monthView, boolean inMonthView,
+                          TaskManager taskManager, int day, int taskIndex, String newDescription)
+        throws TaskManagerException, DateTimeParseException
+```
+
+### Parameters
+
+- `scanner`: Scanner object for user input.
+- `weekView`: Current week being viewed.
+- `monthView`: Current month being viewed.
+- `inMonthView`: Boolean indicating whether the month is being viewed.
+- `taskManager`: TaskManager instance managing tasks.
+- `day`: Day of the task.
+- `taskIndex`: Index of the task to update.
+- `newDescription`: Updated description of the task.
+
+### Exceptions
+
+- `TaskManagerException`: Thrown if not in the correct week/month view.
+- `DateTimeParseException`: Thrown if there is an error parsing the date.
```

Comment:
Not sure if the method parameters are necessary - too much information? - might want to consider removing these